### PR TITLE
fixed #155: back to top button

### LINKED ISSUE #155: Animated back to top button with CSS
The idea is to show a back to top button only after scrolling with CSS.

[This](https://www.freecodecamp.org/news/css-only-back-to-top-button/
) article shows some methods that only use CSS, no JS.

I especially like [this](https://codepen.io/t_afif/pen/ExwbZKQ/86f79c96520a0c864324db70a6b76c76) method with a fading button.

What do you guys think?

### PR DESCRIPTION
**Pull request by mansoorbarri**
Fixes #155

Screenshot: 
image

### CODE REVIEW

**Comment by yashmehrotra** [comment]
@mansoorbarri Thanks for the PR

Can you also:
1. Feature gate it with the `ShowBackToTopButton` param
2. Add the feature in theme-basic-documentation

**Comment by mansoorbarri** [comment]
fixed! in 

> @mansoorbarri Thanks for the PR
> 
> Can you also:
> 
> 1. Feature gate it with the `ShowBackToTopButton` param
> 2. Add the feature in theme-basic-documentation

fixed

File: assets/css/main.css [hunk 1]
```diff
@@ -624,3 +624,22 @@ table td {
...
+  position: fixed; /* Fixed/sticky position */
+  bottom: 20px; /* Place the icon at the bottom of the page */
+  right: 30px; /* Place the icon 30px from the right */
+  z-index: 99; /* Make sure it does not overlap */
+  background-color: transparent; /* Remove the background color */
+  color: var(--accent-color); /* Icon color */
+  cursor: pointer; /* Add a mouse pointer on hover */
+  padding: 10px; /* Adjust the padding as needed */
+  border-radius: 50%; /* Make it a circle */
+  font-size: 24px; /* Increase font size */
```
Comment (526avijitgupta) [inline assets/css/main.css:640]: 1. please remove comments
2. use z-index `5` (to match the `z-index` value of header element)
3. order alphabetically

```css
.arrow-logo {
  background-color: transparent;
  border-radius: 50%;
  bottom: 20px;
  color: var(--accent-color);
  cursor: pointer;
  display: none;
  font-size: 24px;
  padding: 10px;
  position: fixed;
  right: 30px;
  z-index: 5;
}
```

**Review by 526avijitgupta — COMMENTED** [review]
Hi @mansoorbarri thanks for submitting the PR. This feature would be quite useful :)

While the feature currently works, it abruptly takes the user to the top of the page. I would prefer if there is a smooth srolling to the top of the page. What I mean is it should work similar to how clicking on an item in the Table of Contents (on the right side of the page) works. @yashmehrotra wdyt?

Here's a code reference of how to smooth scrolling is implemented currently: https://github.com/526avijitgupta/gokarna/blob/be5611d4f02d679b43e869c1a542ecbf84f0febc/assets/js/main.js#L68C1-L78C2

**Comment by mansoorbarri** [comment]
appreciate the feedback. Please have a look now

**Review by yashmehrotra — APPROVED** [review]
LGTM 👍🏽 

@526avijitgupta Can you have a final look and merge this ?

**Comment by 526avijitgupta** [comment]
@mansoorbarri This seems to break on mobile. . Could you please double check on your side and fix it before we merge?

Otherwise if someone builds Gokarna on the latest version and tries to use it on mobile screens it wouldn't be a good user experience for them

Also, let's use an icon which doesn't have a vertical line at at all. Basically, similar to an inverted "V"

**Comment by yashmehrotra** [comment]
Yeah, we should either pad it in way it does not come over text or disable it for mobile

image

**Comment by mansoorbarri** [comment]
I find this a bit alright but removing the button on mobile is probably the best option in my opinion. Do let me know what you would like.

preview: 
| Dark         | Light          |
| ------------------ | ------------------ |
| image | image|

**Comment by 526avijitgupta** [comment]
@mansoorbarri Thanks for making the changes.
In the current state the button looks okay, but we can certainly make it more visually appealing by making it match the site theme (light/dark).

I was browsing Hugo themes and came across this as an example: https://hugoloveit.com/emoji-support/#

Notice how the button's color and background color matches the current theme (light/dark).

**Comment by 526avijitgupta** [comment]
Basically, what I mean is the blue background color on the back to top button looks a little out of place IMO. Simply changing that to match the light/dark theme should also be enough :)

**Comment by mansoorbarri** [comment]
definitely, my first idea was to make it the same colour as code blocks on the website but I wasn't sure about that. 

preview: https://gokarna.mansoorbarri.com

they look something like this: 
| Dark         | Light      |
| ------------------ | ------------------ |
| image | image |

**Comment by mansoorbarri** [comment]
> definitely, my first idea was to make it the same colour as code blocks on the website but I wasn't sure about that.
> 
> preview: https://gokarna.mansoorbarri.com
> 
> they look something like this:
> 
> Dark	Light
> image	image

we can also make it a bit opaque but if you have any other ideas, do let me know.

**Comment by yashmehrotra** [comment]
@mansoorbarri @526avijitgupta 

I think we should disable this on mobile and create a different issue to track that till we come up with a better design.
Lets try to get this PR merged with mobile disabled

What do you all think ?

**Comment by mansoorbarri** [comment]
I agree

**Comment by mansoorbarri** [comment]
the most we can do is move the button a bit towards right but it still comes over some of the page content: 
image

**Comment by yashmehrotra** [comment]
> the most we can do is move the button a bit towards right but it still comes over some of the page content:

@mansoorbarri Yeah, that is why I don't want to add this in mobile till we come up with a better solution

But I like adding a background color with the arrow

Lets:
1. Disable it for mobile
2. Add background color with arrow with just the inverted v

**Comment by 526avijitgupta** [comment]
> @mansoorbarri @526avijitgupta
> 
> I think we should disable this on mobile and create a different issue to track that till we come up with a better design.

@yashmehrotra Agreed!

**Comment by mansoorbarri** [comment]
> > the most we can do is move the button a bit towards right but it still comes over some of the page content:
> 
> @mansoorbarri Yeah, that is why I don't want to add this in mobile till we come up with a better solution
> 
> But I like adding a background color with the arrow
> 
> Lets:
> 
> 1. Disable it for mobile
> 2. Add background color with arrow with just the inverted v

added

**Comment by yashmehrotra** [comment]
image

Can we make the padding in such a way that the V is in the middle, right now its left-centered

**Comment by mansoorbarri** [comment]
I tried multiple changes to the CSS code however it still stays left aligned, the `svg` doesn't have a seperate div so I don't know why it shows up like that

File: assets/css/main.css [hunk 2]
```diff
@@ -624,3 +624,33 @@ table td {
...
+  bottom: 20px;
+  cursor: pointer;
+  display: none;
+  font-size: 24px;
+  padding: 10px;
+  position: fixed;
+  right: 20px;
+  z-index: 5;
+  background-color: var(--light-tertiary-color);
+  border-radius: 10px;
```
Comment (yashmehrotra) [inline assets/css/main.css:641]: Border radius is twice, lets make this a complete circle

**Comment by yashmehrotra** [comment]
@mansoorbarri Doing `padding-right: 8px` makes it look like its at the centre
Also, lets make this a complete circle

**Comment by mansoorbarri** [comment]
@yashmehrotra all changes are made, preview: https://gokarna.mansoorbarri.com/

**Comment by yashmehrotra** [comment]
@mansoorbarri Looks good

@526avijitgupta Should we merge this PR now ?

**Pull request merged**